p2p/p2p.go

```diff
@@ -67,6 +65,11 @@ func NewTCPNode(cfg Config, key *ecdsa.PrivateKey, connGater ConnGater, opts ...
 		libp2p.AddrsFactory(func([]ma.Multiaddr) []ma.Multiaddr { return nil }),
 	}
 
+	// Set TCP listen addresses.
+	if len(addrs) > 0 {
```

Comment:
don't think this is actually required, see the implementation, it is a noop if len==0